pyladie_oop/pylady.py

```diff
@@ -0,0 +1,103 @@
+class PyLady:
+
+    def __init__(self, name):
+        self.name = name
+        self.knowledge = 0  # max je 5
+        self.max_knowledge = 5
+        self.practice = 0  # max je 5
+        self.max_practice = 5
+        self.focus = 3  # minimum 0
+        self.max_focus = 3
+        self.energy = 5  # minimum 0
+        self.max_energy = 5
+        self.coach_capacity = 3  # min je 0
+        self.max_coach_capacity = 3
+
+    def __str__(self):
+        return f"PyLady {self.name}. Její practice je {self.graficky_practice()} a knowledge {self.graficky_knowledge()}"
+
+    def graficky_knowledge(self):
+        celkem = int(self.max_knowledge)
+        pocet = int(self.knowledge)
+        return "[{0}{1}]".format("#"*pocet, "_" * (celkem-pocet))
+
+    def graficky_practice(self):
+        celkem = int(self.max_practice)
+        pocet = int(self.practice)
+        return "[{0}{1}]".format("#"*pocet, "_" * (celkem-pocet))
+
+    def study(self, hour):
+        # zvyšuje: practice a knowledge podle délky studia
+        # snižuje: energy a focus, obojí nesmí být 0
+        # při energy <= 0 vypíše hlášku
+        # při focus <= 0 vypíše hlášku
+
+        if self.energy > 0 and self.focus > 0:
+
+            hour = int(hour)
+            self.energy -= 1 * hour
+            if self.energy <= 0:
+                self.energy = 0
+
+            self.focus -= 1 * hour
+            if self.focus <= 0:
+                self.focus = 0
+
+            self.knowledge += 1/20 * hour
+            self.practice += 1/10 * hour
+
+    def homework(self):
```

Comment:
Skús sa s tým trochu pohrať, daj si tam ako argument náročnosť a podľa nej vyžaduj knowledge a practice na dokončenie. Môžeš si napr. do initu dať atribut homework s hodnotou 0 a pri dokončení úlohy ho zvyšovať o 1 a mať to ako counter. Zároveň printuj hlášky keď úlohu dokončí, nemá energiu alebo nemá potrebné knowledge/practice. Môžeš si vytvoriť (mimo triedu) napr. slovník s domácimi úlohami, ktorý bude obsahovať ako kľúč názov domácej úlohy (napr. funkcie, triedy, testovanie, proste to čo ste sa učili) a v hodnote to bude obsahovať zanorený slovník kde definuješ tie potrebné hodnoty knowledge a practice pre danú úlohu.

Napr. `homework_requirements = {'testing': {'knowledge': 2, 'practice': 1.5}, 'turtle': {'knowledge': 0.5, 'practice': 0.3}}`